binr/r2agent/r2agent.c

```diff
@@ -104,10 +120,13 @@ int main(int argc, char **argv) {
 		char *result_heap = NULL;
 		const char *result = page_index;
 
-		rs = r_socket_http_accept (s, 0, timeout);
+		rs = r_socket_http_accept (s, 0, timeout, httpauth, authtoken);
```

Comment:
i think it will be better to refactor this into RSocketHttpOptions struct and put the timeout, and other configuration options in there.

or maybe we can just add a callback which is used to process the authentication check.

Reply:
otherwise we cant have more than one user, also will be good to define privilege levels to different users and make r2 multiuser

Reply:
That s true it s better having multi user even though as stated if the original PR comment, it was purposely mono-user as a "first step" but will try to come up with something along ...